src/chapters/enums/enums-in-models.rst

```diff
@@ -14,3 +14,241 @@ The final code from this video is in the `enums branch <https://github.com/Launc
+      this();
+      this.name = name;
+      this.description = description;
+      this.contactEmail = contactEmail;
+      this.type = type;
+   }
+
+   public EventType getType() {
+      return type;
+   }
+
+   public void setType(EventType type) {
+      this.type = type;
+   }
+
+.. _passing-enums-through-controller:
+
+Pass Enum Values Through the Controller
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``coding-events`` uses model binding to create an ``Event`` object. So like any other field on 
+the model, the controller does not necessarily need to know about the addition of ``Event.type`` 
+in order to create an ``Event`` instance from a form. However, we want the user to choose from 
+the pre-defined event type values when creating their event. To do this, we'll use the 
+controller method ``displayCreateEventForm`` to pass those values into the view.
+
+In ``EventController``:
+
+.. sourcecode:: java
+   :lineno-start: 26
+
+   @GetMapping("create")
+   public String displayCreateEventForm(Model model) {
+      model.addAttribute("title", "Create Event");
+      model.addAttribute(new Event());
+      model.addAttribute("types", EventType.values());
+      return "events/create";
+   }
+
+``.values()`` is a built-in static method that returns an array of values defined in 
+the given enum, in the order in which they have been declared.
+
+With the template variable ``types`` now defined, we can use our ``EventType`` values in the view.
+
+Use Enum Value in a ``Select`` Element
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The list of constants returned from ``EventType`` lends itself well to a ``select``-type form 
+input. We'll update our form so that a user will have the option to choose one of the provided 
+event types from a dropdown menu.
+
+In ``templates/events/create.html``:
+
+.. sourcecode:: html
+   :lineno-start: 27
+
+   <div class="form-group">
+      <label> Type
+      <select th:field="${event.type}">
+         <option th:each="type : ${types}"
+                  th:value="${type}"
+                  th:text="${type.displayName}"
+         ></option>
+      </select>
+      </label>
+   </div>
+
+As with the other form inputs on the page, the ``th:field`` attribute determines the ``name``
+and ``id`` attributes for the ``select`` tag. We make an ``option`` tag for each of the ``EventType``
+values, making use of the ``types`` variable we passed in from the controller in 
+:ref:`the previous step <passing-enums-through-controller>`. We set the ``value`` attribute for the 
+model data to be the ``EventType`` value using ``th:value``. And the type name shown to the user 
+of the form as the ``displayName`` of the type, using ``th:text``.
+
+Use Enum Properties to Display Information
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once an event is created, to display its ``type`` field in the table of all events, we'll modify 
+``templates/events/index.html`` to include another column:
+
+.. sourcecode:: html
```

Comment:
Line numbers feel odd on this example, since we're implicitly skipping several lines between the two new snippets.